src/SmartComponents/ActivityTable/Columns.js

```diff
@@ -17,28 +23,140 @@ TaskNameCell.propTypes = {
   index: propTypes.number,
 };
 
-const StatusCell = ({ status }) =>
-  ({
-    Completed: (
-      <span style={{ color: '#3E8635', display: 'flex', alignItems: 'center' }}>
-        <Icon status="success" style={{ marginRight: '4px' }}>
-          <CheckCircleIcon />
-        </Icon>
-        {status}
-      </span>
-    ),
-    Running: (
-      <span style={{ color: '#2B9AF3', display: 'flex', alignItems: 'center' }}>
-        <Icon status="info" style={{ marginRight: '4px' }}>
-          <InProgressIcon />
-        </Icon>
-        {status}
-      </span>
-    ),
-  }[status] || status);
+const getStatusProps = (task) => {
+  let badgeColor, iconStatus, IconType, popoverBodyContent, hasPopover;
+
+  popoverBodyContent = '';
+  hasPopover = true;
+  switch (task.status) {
+    case TASK_STATUS.RUNNING:
+      badgeColor = '#2B9AF3';
+      iconStatus = 'info';
+      IconType = InProgressIcon;
+      popoverBodyContent = [
+        `${task.running_jobs_count || 0} Running`,
+        `${task.completed_jobs_count || 0} Completed`,
+      ];
+      break;
+    case TASK_STATUS.COMPLETED:
+      badgeColor = '#3E8635';
+      iconStatus = 'success';
+      IconType = CheckCircleIcon;
+      hasPopover = false;
+      break;
+    case TASK_STATUS.COMPLETED_WITH_ERRORS:
+      badgeColor = '#FF0000';
+      iconStatus = 'danger';
+      IconType = ExclamationCircleIcon;
+      popoverBodyContent = [
+        `${task.failure_jobs_count || 0} Failed`,
+        `${task.timeout_jobs_count || 0} Timeout`,
+        `${task.running_jobs_count || 0} Running`,
+        `${task.completed_jobs_count || 0} Completed`,
+      ];
+      break;
+    case TASK_STATUS.FAILURE:
+      badgeColor = '#FF0000';
+      iconStatus = 'danger';
+      IconType = ExclamationCircleIcon;
+      popoverBodyContent =
+        'All jobs in this task failed to run on their target systems.';
+      break;
+    case TASK_STATUS.CANCELLED:
+      badgeColor = '#FF0000';
+      iconStatus = 'danger';
+      IconType = ExclamationCircleIcon;
+      popoverBodyContent = 'All jobs in this task have been cancelled';
+      break;
+  }
+
+  return { badgeColor, iconStatus, IconType, popoverBodyContent, hasPopover };
+};
```

Comment:
I think you can simply this like so:

```
const getStatusProps = (task) => {
  switch (task.status) {
    case TASK_STATUS.RUNNING:
      return {
        badgeColor: '#2B9AF3',
        iconStatus: 'info',
        IconType: InProgressIcon,
        popoverBodyContent: [
          `${task.running_jobs_count || 0} Running`,
          `${task.completed_jobs_count || 0} Completed`,
        ],
        hasPopover: true,
      };
    case TASK_STATUS.COMPLETED:
      return {
        badgeColor: '#3E8635',
        iconStatus: 'success',
        IconType: CheckCircleIcon,
        hasPopover: false,
      };
    case TASK_STATUS.COMPLETED_WITH_ERRORS:
      return {
        badgeColor: '#FF0000',
        iconStatus: 'danger',
        IconType: ExclamationCircleIcon,
        popoverBodyContent: [
          `${task.failure_jobs_count || 0} Failed`,
          `${task.timeout_jobs_count || 0} Timeout`,
          `${task.running_jobs_count || 0} Running`,
          `${task.completed_jobs_count || 0} Completed`,
        ],
        hasPopover: true,
      };
    case TASK_STATUS.FAILURE:
      return {
        badgeColor: '#FF0000',
        iconStatus: 'danger',
        IconType: ExclamationCircleIcon,
        popoverBodyContent:
          'All jobs in this task failed to run on their target systems.',
        hasPopover: true,
      };
    case TASK_STATUS.CANCELLED:
      return {
        badgeColor: '#FF0000',
        iconStatus: 'danger',
        IconType: ExclamationCircleIcon,
        popoverBodyContent: 'All jobs in this task have been cancelled',
        hasPopover: true,
      };
  }
};
```

Reply:
generally speaking, we want to avoid setting variables with `let` when possible. It's not always possible, but here I think is a good trade off.